src/lib.ml

```diff
@@ -803,6 +806,13 @@ let rec instantiate db name (info : Lib_info.t) ~stack ~hidden =
   in
   let default_implementation =
     Option.map info.default_implementation ~f:(fun l -> lazy (resolve l)) in
+  let implementations =
+    Option.map info.virtual_ ~f:(fun _ -> lazy (
+      let available_implementations = db.find_implementations name in
+      Variant.Map.map available_implementations ~f:(
+        List.map ~f:(fun (impl : Lib_info.t) ->
+          resolve (impl.loc, impl.name)))))
+  in
```

Comment:
It looks like implementations are not searched recursively through databases. I think it should either be done here when `available_implementations` is empty, or in `db.find_implementations`.

Reply:
Yeah, I was a bit suspicious but I thought our test suite covered this case. Can you think of a test case that would trigger this back?

Reply:
Also, why is it ok to do only look recursively when available_implementations are empty? Shouldn't we consider implementations in the parent db regardless.

Reply:
> Can you think of a test case that would trigger this back?

* Putting the implementation and the virtual library in different scopes should trigger the recursive seek. We can do that using several `dune-project` files, right ?
* It can also be triggered if the implementation lives in an installed opam package but this seems difficult to setup as a test.

> Also, why is it ok to do only look recursively when available_implementations are empty? 

I was afraid one implementation living both in a local directory and in the opam scope would conflict. But it looks like your recursive lookup of implementations takes care of this !